koji_promote: report all pushed images

twaugh opened the pull request:
When reporting pushed images, don't just look at the results of the `pulp_push` plugin.

For v1: report only `pulp_push` results
For v1+v2: report `pulp_push` and `pulp_sync` results
For v2: report only `pulp_sync` results

As osbs-client will configure pulp_push and pulp_sync according to whether v1 and/or v2 is in use, all we need to do is collect the results from both pulp plugins.

If pulp is not used, report `tag_and_push` results (now modifed to return `ImageName` instances).

jpopelka commented:
Shouldn't the `TestKojiPromote` be now skipped also `if PULP_SYNC_KEY is None` ?
i.e. 
```
@pytest.mark.skipif(PULP_PUSH_KEY is None or PULP_SYNC_KEY is None,
                    reason="requires pulp_push and pulp_sync")
class TestKojiPromote(object):
```

twaugh commented:
Oh, true. In practice they'll either both be or neither, but I'll change it anyway.

jpopelka commented:
LGTM

TomasTomecek commented:
what about metadata exit plugin?

twaugh commented:
Because `pulp_sync` adds to the `PushConf` for the workflow, the metadata plugin picks it up automatically.

...which suggests that's how `koji_promote` ought to be doing it as well.

TomasTomecek commented:
@twaugh thanks Tim; LGTM then

twaugh commented:
I've updated the `koji_promote` plugin to use TagConf+PushConf instead of looking through plugin results.

I also added two other fixes in separate commits: I think the metadata "name" field needs to include the image's namespace as well, and the tests did not run correctly in Python 3.

jpopelka commented:
LGTM

twaugh commented:
Thanks. Re-basing and squashing fix-fix commits.

TomasTomecek commented:
LGTM
